### PR TITLE
[Fix] optimize visual token mask with caching and multi-token support

### PR DESCRIPTION
**Pull request by bo-ke**
## Purpose
Optimize `_set_visual_token_mask` function to improve performance and extend functionality for multimodal models.

**Changes:**
- Add caching mechanism for visual token IDs to reduce repeated list creation overhead
- Extend support to handle multiple visual token types (image/video start/end tokens, not just `im_patch_id`)
- Replace multiple equality comparisons with `torch.isin` for efficient multi-token matching

### CODE REVIEW

File: vllm/model_executor/models/ernie45_vl.py [hunk 1]
```diff
@@ -1404,7 +1404,7 @@ def _set_visual_token_mask(self, input_ids: torch.Tensor) -> None:
             return
 
         # Cache the visual token IDs tensor to avoid recreating it
-        if not hasattr(self, "_visual_token_ids_cache"):
+        if not hasattr(self, "_visual_token_ids_tensor_cache"):
```
Comment (Isotr0py) [inline vllm/model_executor/models/ernie45_vl.py]: I think we can directly initialize `self._visual_token_ids_tensor_cache` in `__init__` to simplify this function.

File: vllm/model_executor/models/ernie45_vl.py [hunk 2]
```diff
@@ -1398,12 +1398,36 @@ def _vision_forward(
...
             self.visual_token_mask = None
+            return
+
+        # Cache the visual token IDs tensor to avoid recreating it
+        if not hasattr(self, "_visual_token_ids_tensor_cache"):
+            visual_token_ids = [
+                token_id for token_id in [
+                    self.config.im_patch_id,
+                    getattr(self.config, "image_start_token_id", None),
+                    getattr(self.config, "image_end_token_id", None),
```
Comment (Isotr0py) [inline vllm/model_executor/models/ernie45_vl.py]: Seems the `image_start_token_id` and `image_end_token_id` don't appear in `config.json`? (https://huggingface.co/baidu/ERNIE-4.5-VL-28B-A3B-PT/blob/main/config.json)

**Review by Isotr0py — COMMENTED** [review]
Look reasonable to me. Just have some nits.

File: vllm/model_executor/models/ernie45_vl.py [hunk 3]
```diff
@@ -1398,12 +1398,36 @@ def _vision_forward(
...
+                ] if token_id is not None
+            ]
+            self._visual_token_ids_tensor_cache = torch.tensor(
+                visual_token_ids, dtype=torch.long)
+
+        # Create tensor on the correct device
+        visual_token_ids_tensor = self._visual_token_ids_tensor_cache.to(
+            device=input_ids.device,
+            dtype=input_ids.dtype,
+        )
```
Comment (ywang96) [inline vllm/model_executor/models/ernie45_vl.py]: This seems a static tensor - why don't we simply set it at init time?

Comment (bo-ke) [reply vllm/model_executor/models/ernie45_vl.py]: great suggestion

File: vllm/model_executor/models/ernie45_vl.py (hunk 2 above)
Comment (CSWYF3634076) [reply vllm/model_executor/models/ernie45_vl.py]: The relevant model configurations have been updated
https://huggingface.co/baidu/ERNIE-4.5-VL-28B-A3B-PT/commit/41a0c75292ebc34d0ff143457353505646fd8d92

**Review by Isotr0py — APPROVED** [review]
LGTM now!
